### PR TITLE
feat: [SRM-12743]: Updated delete API

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/__tests__/CVCreateSLOV2.test.tsx [hunk 1]
```diff
@@ -65,6 +65,10 @@ jest.mock('services/cv', () => ({
 }))
 
 describe('CVCreateSloV2', () => {
+  beforeEach(() => {
+    jest.setTimeout(30000)
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/__tests__/CVCreateSLOV2.test.tsx:69]: Why we are having this timeout?

Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/__tests__/CVCreateSLOV2.test.tsx:69]: jest was failing with  Async callback was not invoked within the 10000 ms timeout specified by, so increased

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/slos/CVSLODetailsPage/CVSLODetailsPage.tsx [hunk 2]
```diff
@@ -89,6 +90,14 @@ const CVSLODetailsPage: React.FC = () => {
...
   })
 
+  const { mutate: deleteSLOV2, loading: deleteSLOV2Loading } = useDeleteSLOV2Data({
+    queryParams: {
+      accountId,
+      orgIdentifier,
+      projectIdentifier
+    }
+  })
+
```
Comment (karansaraswat19) [inline src/modules/85-cv/pages/slos/CVSLODetailsPage/CVSLODetailsPage.tsx:100]: Why are we using useDeleteSLOV2Data and useDeleteSLOData both? Deletion can be done by the new api for both the types.

Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/CVSLODetailsPage/CVSLODetailsPage.tsx:100]: keep it backward compatible as this is under feature flag

**Review by karansaraswat19 — APPROVED** [review]

**Pull request merged**